### DISCUSSION TITLE
HA Integration

### DISCUSSION THREAD

**Discussion by deliverer33** [MartinStoffel/tigo#2] [General]
Does this require the paid API service?  Also, does it show each individual panel output?

**Comment by MartinStoffel** [comment]
Yes, you need a tigo premium license to access the api. The values shown on the screenshot are per panel/tigo ts4.
I hope this answers your questions.

**Reply by MartinStoffel** [reply]
Just made a change, so you get the values for Energy, Power and Reclaimed Power if you only have the default license.

**Reply by deliverer33** [reply]
Awesome I will give it a try.  I definitely want the data in HA so I will try the free version then upgrade to paid version

**Reply by deliverer33** [reply]
Does the paid api allow you to pull data per panel like the mobile app displays?  That might be one suggestion to add.

**Reply by MartinStoffel** [reply]
> Does the paid api allow you to pull data per panel like the mobile app displays? That might be one suggestion to add.

I'm not sure if I understood your suggestion. The integration gives you already the data per panel and stores it then in HA. So there is no real need to query it.

**Reply by deliverer33** [reply]
Sorry if I'm asking dumb questions, I'm new to Tigo and solar in general.  I did set it up successfully and I'm getting data into HA now.  I'm currently on the free/unpaid version of the API.  Is there an explanation of the values and what they map to?  Example, A,B,C....Power, Reclaimed Power.  I'm kind of surprised I'm getting data for each of these with the free API?

**Reply by deliverer33** [reply]
Are these mapping to my strings?

**Reply by MartinStoffel** [reply]
No problem, with this integration you see basically what you also see in the Tigo App on the panel level. Hence A1, is the first panel on string A, A2 the second and so on. 

Power is the current power the panel delivers, Reclaimed Power is what you gain due to the fact, that you use Tigo optimizers and Energy is the energy produced by each panel (Integration of power over time).

If you have the paid API you will get in addition: Voltage In/Out, Temperature, Current In, RSSI (signal strength) and PWM (pulse-width modulation).

**Reply by deliverer33** [reply]
Awesome work!  Now I just need to figure out how to feed the data into something like Grafana.  I think the community is really going to appreciate this integration.  I know I do.

**Comment by Koky05** [comment]
Can you agregate received data specific objects?
Sorry for too big suggestion.
I will change initialization load just for login and password  to obtain bearer (/api/user/login). 
In next step it will call for list of systems (/api/v3/systems) and prompt user to choose which system he wants to integrate (or just take all system_id's).
After that steps it will call system objects (/api/v3/objects/system) and (/api/v3/objects/types) to load list of all available object in systems with apropriate types (System/Panel/String/Inverter).
With all that date will be created device as per system_id and under it it will be created entities like now or object with state as power and all other relevant values will be as attributes. 

I'm just starting with python after som years with java and VBA. But if I can help with that somehow I will.

**Reply by MartinStoffel** [reply]
Hi  @Koky05 

Thank you for your input. Since most users have not more than one Tigo system I would like to keep it as it is. In addition, this integration uses a reverse engineered interface mostly - not the api which is behind a pay wall. Therefore not everything possible with the API can be done.
